File: backend/interfaces/api/v1/schemas/baggage.py
"""수화물 체커 API 스키마."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field, ConfigDict


class BaggageCheckRequest(BaseModel):
    """수화물 판정 요청 스키마."""

    airline: str = Field(
        ...,
        min_length=1,
        max_length=100,
        description="항공사명 (예: '대한항공', '아시아나')"
    )
    product: str = Field(
        ...,
        min_length=1,
        max_length=255,
        description="제품명 (예: '맥북', '보조배터리', '우산')"
    )
    value: float | None = Field(
        None,
        ge=0,
        description="수량/크기 (선택사항)"
    )
    unit: str | None = Field(
        None,
        description="단위 (선택사항, inch, cm, kg, g, mg, oz, lb)"
    )

    model_config = ConfigDict(from_attributes=True)


class VerdictResponse(BaseModel):
    """판정 결과 응답 스키마."""

    verdict: str = Field(..., description="판정 타입 (allowed, conditional, forbidden)")
    label: str = Field(..., description="한국어 라벨")
    reason: str = Field(..., description="판정 이유")
    emoji: str = Field(..., description="이모지")
    color_code: str = Field(..., description="HTML 색상 코드")
    is_allowed: bool = Field(..., description="허용 여부")
    is_forbidden: bool = Field(..., description="불가 여부")

    @classmethod
    def from_domain(cls, verdict) -> "VerdictResponse":
        """도메인 Verdict에서 변환."""
        return cls(
            verdict=verdict.verdict.value,
            label=verdict.label,
            reason=verdict.reason,
            emoji=verdict.emoji,
            color_code=verdict.color_code,
            is_allowed=verdict.is_allowed,
            is_forbidden=verdict.is_forbidden,
        )


class BaggageCheckResponse(BaseModel):
    """수화물 판정 응답 스키마."""

    carry_on: VerdictResponse = Field(..., description="기내 반입 판정")
    checked: VerdictResponse = Field(..., description="위탁물로 판정")
    checked_at: datetime = Field(default_factory=datetime.utcnow, description="체크 시간")

    model_config = ConfigDict(from_attributes=True)

File: frontend/lib/api/baggage.ts
/**
 * 수화물 체커 관련 API
 */
import { apiPost } from '../api-client';

export interface BaggageCheckRequest {
  airline: string;
  product: string;
  value: number;
  unit: 'ml' | 'g' | 'l' | 'kg' | 'inch' | 'cm';
}

/**
 * 백엔드 VerdictType(app/domain/value_objects/verdict.py)과 1:1 대응.
 * globals.css의 `.verdict.allowed / .conditional / .forbidden`도 이 값을 사용한다.
 */
export type VerdictType = 'allowed' | 'conditional' | 'forbidden';

export interface Verdict {
  verdict: VerdictType;
  label: string;
  reason: string;
  emoji?: string;
  color_code?: string;
  is_allowed: boolean;
  is_forbidden: boolean;
}

export interface BaggageCheckResponse {
  carry_on: Verdict;
  checked: Verdict;
  checked_at: string;
}

/**
 * 수화물 규정 체크
 */
export async function checkBaggage(
  request: BaggageCheckRequest,
  accessToken?: string
): Promise<BaggageCheckResponse> {
  return apiPost<BaggageCheckResponse>(
    '/api/v1/baggage/check',
    request,
    accessToken
  );
}